c_glib/arrow-glib/decimal.cpp

```diff
@@ -60,11 +60,18 @@ struct DecimalConverter<arrow::Decimal256> {
 
 template <typename Decimal>
 typename DecimalConverter<Decimal>::GArrowType *
-garrow_decimal_new_string(const gchar *data)
-{
-  auto arrow_decimal = std::make_shared<Decimal>(data);
-  DecimalConverter<Decimal> converter;
-  return converter.new_raw(&arrow_decimal);
+garrow_decimal_new_string(const gchar *data,
+                          GError **error,
+                          const gchar *tag)
+{
+  auto arrow_decimal_result = Decimal::FromString(data);
+  if (garrow::check(error, arrow_decimal_result, tag)) {
+    auto arrow_decimal = std::make_shared<Decimal>(*arrow_decimal_result);
+    DecimalConverter<Decimal> converter;
+    return converter.new_raw(&arrow_decimal);
+  } else {
+    return nullptr;
+  }
```

Comment:
Good catch.
Generally, I use `NULL` for GLib (C) related API. This should be this case because we just returns the return value of this template function in GLib related API. I'll change to use `NULL`.